### PR TITLE
[READY]Major explosive rebalanance and changes patch: Edge strikes back

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request

This PR rebalances the explosives. 

Changes:
- TaTP - TaTP is now the most powerful **large scale** (>1000 u) explosive chemical. The more units you use the lower strengthdivider becomes, topping off at 1.5 at around 2100u. Which is 25% better than nitroglycerin.

- RDX - RDX now produces a slightly bigger fire wave. I felt like the chem needed a small buff due to it being hard to craft. It now requires gold as a catalyst to craft. Prevents the use of macros since you now gotta get some gold.

- Methamphetamine - Meth was nerfed from strengthdiv of 6 to 12, modifier changed from 1 to 5. This won't affect small scale explosions that much, it will still destroy your meth lab, but bigger explosions will be underwhelming. I felt like meth was the go-to braindead explosive, since pretty much everyone has macro's for it.

- Blackpowder - nerfed strengthdiv from 6 to 8, buffed modifier from 2 to 5. Same reason as Meth, it is very easy to craft , so it will be weaker.

- Nitric acid - I have recently checkced the formula and noticed that fluorosulphuric acid is better than nitric acid in every single way! I have decided to move some power from facid to nacid, because nacid is an upgrade, so it should be better. Nerfed facid's fireloss power and moved it to nitric acid. Death syringes are still not possible.

- Exotic Stabilizer - created with stabilizing agent and hyper-plasmium oxide

- Hyper-plasmium oxide - commonly found on lavaland in geysers.

Additionally i made chemical_reactions AUTODOC compliant.

## Why It's Good For The Game

I felt like explosives were too easy to craft. With this rebalance i made a big seperation - higher tier explosives that can really deal a lot of damage should be hard to create or require cross-dep collaboration. Lower tier explosives should be a lot weaker, to prohibit people from abusing macro's to pump out tons of high-tier explosives.

Availability of high-tier explosives should be limited to chemists.

TL;DR meth is now worse, tatp is better, you can now make ghetto blackpowder.

## Changelog
:cl:
balance: Nitroglycerin : You now have to use exotic stabilizer.
balance: TaTP : The more you use it, the more powerful it becomes. You now have to stabilize it with exotic stabilizer
balance: RDX : produces slightly bigger heatwave and slightly bigger shockwave. Now requires gold.
balance: Gunpowder : strengthdiv 8 - > 10 (nerf), modifier 2 - > 5 (buff).
balance: Methamphetamine : strengthdiv 6 -> 12 (nerf), modifier 1 - > 5 (buff).
balance: Moved some power from facid to nitric acid.
add: Hyper-plasmium oxide can be commonly found in lavaland geysers
add: Exotic stabilizer can now be crafted with hyper-plasmium oxide and stabilizing agent
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 1]
```diff
@@ -115,6 +115,10 @@
 	e.start()
 	holder.clear_reagents()
 
+/datum/reagent/gunpowder/ghetto
+	name = "Improvised Gunpowder"
+	description = "Better than nothing i guess."
```
Comment (Fikou) [inline code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:120]: capitalize your i's BiTCH

Comment (EdgeLordExe) [reply code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:120]: alright fIkou

**Comment by optimumtact** [comment]
This seems like a reasonable reset of the tiers.

Can you maxcap with TATP?

**Comment by EdgeLordExe** [comment]
> This seems like a reasonable reset of the tiers.
> 
> Can you maxcap with TATP?

you can already do that with nitroglycerin/tatp. Chemical explosions are not subject to TTV maxcap

**Comment by optimumtact** [comment]
so they're bigger than bombs?

That shit aint gonna fly, they should scale to the maxcap dynamically.

**Comment by EdgeLordExe** [comment]
> so they're bigger than bombs?
> 
> That shit aint gonna fly, they should scale to the maxcap dynamically.

well thats a fault with chemical explosion code as a whole. You should also consider the fact that aquiring large amounts of chemicals is much tougher than filling a tank with plasma and oxy. Yes i agree there should be a maxcap that is maybe bigger than the TTV one? , but that is a job for a refactor of the explosion datums, which is out of scope for this pr.

**Comment by EdgeLordExe** [comment]
To even achieve a >maxcap result you have to (as nonantag) 
1. go to lavaland
2. get gibtonite
3. get 12 BS beakers
4. get 10 plasteel
5. create a bomb shell
6. make a chemical payload
7. aquire **2100**u of your chosen chem heated up perfectly under it's explosion temperature
8. put it in the payload and arm the thing int the bomb shell.

Thats a lot more effort than clikcing a few times on a tank and making a maxcap ttv

**Comment by ExcessiveUseOfCobblestone** [comment]
> I felt like the chem needed a small buff due to it being hard to craft.

which part of it cannot be macro'd? 

There is no such thing as hard to craft reaction if it **can** be macro'd via a dispenser. it could be hardER to craft than a reagent made from base chems, but it is not hard.

No new explosives please, explosions are cringe. If you want a ghetto explosion, pick up potas/water or drag a welding fuel tank.

**Comment by EdgeLordExe** [comment]
> > I felt like the chem needed a small buff due to it being hard to craft.
> 
> which part of it cannot be macro'd?
> 
> There is no such thing as hard to craft reaction if it **can** be macro'd via a dispenser. it could be hardER to craft than a reagent made from base chems, but it is not hard.
> 
> No new explosives please, explosions are cringe. If you want a ghetto explosion, pick up potas/water or drag a welding fuel tank.

actually yeah i agree, the ghetto explosive might be an issue. RDX? the fact that you have to make 2 complicated chemical instead of 1 . RDX is shit if used without teslium. so you have to take that into account aswell

**Comment by ExcessiveUseOfCobblestone** [comment]
there is no such thing as a complicated chemical if it can be made purely from the dispenser.

**Comment by EdgeLordExe** [comment]
> there is no such thing as a complicated chemical if it can be made purely from the dispenser.

RDX and Teslium cannot be made purely in the dispenser, unlike meth

**Comment by ExcessiveUseOfCobblestone** [comment]
what component are you referring to for both?

**Comment by EdgeLordExe** [comment]
> what component are you referring to for both?

nitric acid and multiver, unless i missed something

**Comment by ExcessiveUseOfCobblestone** [comment]
you can make both from the dispenser but both must be heated.

**Comment by EdgeLordExe** [comment]
> you can make both from the dispenser but both must be heated.

yeahthey cannot be made purely in dispenser. Other chem sources are quire rare actually, it wouldnt really make sense for a upper-mid tier explosive to require them

**Comment by ExcessiveUseOfCobblestone** [comment]
> RDX and Teslium cannot be made purely in the dispenser, unlike meth(not counting the finishing heating since all the chems are there already)

> because they require heating they cannot be made purely in the dispenser

really bro...

**Comment by EdgeLordExe** [comment]
> > RDX and Teslium cannot be made purely in the dispenser, unlike meth(not counting the finishing heating since all the chems are there already)
> 
> > because they require heating they cannot be made purely in the dispenser
> 
> really bro...

alright alright. That doesnt change meth isnt brain dead easy to make tho

**Comment by ExcessiveUseOfCobblestone** [comment]
It is, and so is every other chem that can be made purely from reagents found in the dispenser (including intermediates that are made from reagents purely from the dispenser) which is why I took issue with RDX becoming some 'higher tier' when the only difference between that and meth is the amount of buttons you click.

**Comment by EdgeLordExe** [comment]
> It is, and so is every other chem that can be made purely from reagents found in the dispenser (including intermediates that are made from reagents purely from the dispenser) which is why I took issue with RDX becoming some 'higher tier' when the only difference between that and meth is the amount of buttons you click.

i guess that's true, but that's due to the fact that there aren't many mechanics that can be utilized in chemistry. I mean, other than geysers and hydroponics what other **unique** reagent sources are found in the game? I'm 100% for adding more of these, and i might even do so in near future, but this is just a rebalance PR, way out of scope to add anything like that.

**Comment by optimumtact** [comment]
they are all in botany

**Comment by EdgeLordExe** [comment]
> they are all in botany

unfortunately

**Comment by optimumtact** [comment]
I told you i'm merging the two departments

**Comment by EdgeLordExe** [comment]
> I told you i'm merging the two departments

Hydroponics after the rework is quite incompatible with current chemistry, the only thing that they share is that they produce chemicals.

**Comment by EdgeLordExe** [comment]
> I told you i'm merging the two departments

y'know what, go ahead, iwant to see this

**Comment by wesoda25** [comment]
oh boy another one of these

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm]: unironically all of your prs have terrible grammar could you please review this stuff before you open it?

Comment (EdgeLordExe) [reply code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm]: unironically im not a native speaker, PR's are for a reason y'know?

**Comment by ArcaneDefence** [comment]
Currently, with nitro I can make an 11/22/45 explosion via chem payload at 2100u. 
Do you *REALLY* want to make something that you would describe as "Which is twice as good as nitroglycerin."?

**Comment by EdgeLordExe** [comment]
> Currently, with nitro I can make an 11/22/45 explosion via chem payload at 2100u.
> Do you _REALLY_ want to make something that you would describe as "Which is twice as good as nitroglycerin."?

i well know what is possible within chemistry

**Comment by EdgeLordExe** [comment]
After thinking about it i decided to nerf RDX's reagent by making it require gold as a catalyst

**Comment by EdgeLordExe** [comment]
Also moved some power from facid to nacid, since nacid was for some reason worse in every single way?

**Comment by Pugie** [comment]
doesn't rdx already have catalysts in the form of stabilising agent why does it need more

**Comment by evilcyborgmrt** [comment]
Why not just remove all the explosive chem recipes? Non-antags shouldn't have access to explosives anyway. So let's just cut to the chase and make all explosives available via uplink only.

**Comment by EdgeLordExe** [comment]
> doesn't rdx already have catalysts in the form of stabilising agent why does it need more

stabilizing agent just makes rdx not explode

**Comment by MarioWizard119** [comment]
This just serves to make the new explosives absurdly overpowered while making the ones that are already there have the power of a wet napkin.

The reason meth is so reliable of an explosive is because of how explosives are calculated. Heat activated explosives will always have a bigger yield than ones that are activated on mix. 200u of a heat activated explosive will have 200u of “explosion,” and that’s what the created volume counts as. A mix activated explosive, 100u each of two ingredients will have only 100u of “explosion,” not 200u. Unit for unit, the less ingredients an explosion requires, the stronger it will be. This can be observed with a diethylamine bomb and an overheated meth bomb with the same volume. They both have the same strength divider and modifiers, but the diethylamine bomb explodes at half power. The less ingredients an explosion needs to activate, the stronger it will be. If you want to topple meth for reliability, make it to where all explosives have the same volume regardless of the number of ingredients (ex. a diethylamine bomb would have the same yield as an overheated meth bomb).

Now with that said, RDX has a strength divider of 8 if heated, 4 if you use teslium, soooo unit for unit, it’s still 8, but it’s better with low volume explosives.

You can maxcap with tatp at 700u (and come pretty close at 600u). This may seem like a lot, but you forgot plumbing exists. If it can be crafted with roundstart chemicals, and the only thing not from the dispenser being a catalyst, plumbing can crank out thousands of units of it. This can easily be accomplished with 3 blue space beakers or 6 normal beakers, well within the limits of a chemical payload with common components. Obtaining gibtonite isn’t that much of an issue if you want it that badly, since payloads can fit in backpacks, make the payload at lavaland instead of toting the gibtonite to the station. It’s even more accessible if you’re a traitor and have c4 since you can buy up to 24 sticks of the devil’s playdoh.

Tl;dr: Buff your crappy explosives, don’t nerf the ones that are there. Mass producible maxcaps, not exactly a good time eh?

**Comment by EdgeLordExe** [comment]
> This just serves to make the new explosives absurdly overpowered while making the ones that are already there have the power of a wet napkin.
> 
> The reason meth is so reliable of an explosive is because of how explosives are calculated. Heat activated explosives will always have a bigger yield than ones that are activated on mix. 200u of a heat activated explosive will have 200u of “explosion,” and that’s what the created volume counts as. A mix activated explosive, 100u each of two ingredients will have only 100u of “explosion,” not 200u. Unit for unit, the less ingredients an explosion requires, the stronger it will be. This can be observed with a diethylamine bomb and an overheated meth bomb with the same volume. They both have the same strength divider and modifiers, but the diethylamine bomb explodes at half power. The less ingredients an explosion needs to activate, the stronger it will be. If you want to topple meth for reliability, make it to where all explosives have the same volume regardless of the number of ingredients (ex. a diethylamine bomb would have the same yield as an overheated meth bomb).

Meth from the concept was just supposed to be a hazard while you are making objectively the best drug in the game, but because the mad fucker didnt read the formula or was oblivious to what he was doing, he created a go-to chem for mass destruction.

> 
> Now with that said, RDX has a strength divider of 8 if heated, 4 if you use teslium, soooo unit for unit, it’s still 8, but it’s better with low volume explosives.

Why do you think it is strengthdiv 4 instead of 8?

> 
> You can maxcap with tatp at 700u (and come pretty close at 600u). This may seem like a lot, but you forgot plumbing exists. If it can be crafted with roundstart chemicals, and the only thing not from the dispenser being a catalyst, plumbing can crank out thousands of units of it. This can easily be accomplished with 3 blue space beakers or 6 normal beakers, well within the limits of a chemical payload with common components. Obtaining gibtonite isn’t that much of an issue if you want it that badly, since payloads can fit in backpacks, make the payload at lavaland instead of toting the gibtonite to the station. It’s even more accessible if you’re a traitor and have c4 since you can buy up to 24 sticks of the devil’s playdoh.

TaTP is a gimmick chem that is randomized roundstart preventing the chemist from just simply making it at roundstart.

**Comment by wesoda25** [comment]
the fact of the matter is that within chemistry you can't try and say that difficulty is an obstacle because when - not if - people overcome it, they'll just abuse it and kill others unfairly

**Comment by EdgeLordExe** [comment]
> the fact of the matter is that within chemistry you can't try and say that difficulty is an obstacle because when - not if - people overcome it, they'll just abuse it and kill others unfairly

god chemistry really needs a rework. It is shallow as fuck.

**Comment by halitosisman** [comment]
Experienced chem/explosive manufacturer powergamer here. Perhaps some insight about the current explosive production mechanics in practice would be appreciated. 

For a botany - chemistry partnership to produce a single good nitro bomb within the time of a typical round, a number of requirements need to be satisfied.
-The botanist and chemist must be blood brothers, or must both have sufficient imagination to understand the multitude of problems that high yield ordnance solves for non antags. 
-The duo must be good at their jobs.
-The duo must be uninterrupted in their work. 
-Cargo must be competent enough to print small quantities of simple items at their autolathe for those who ask. 

You're wasting your time trying bothering with nitro grenades. These conditions almost never happen. At the moment they are code bloat. 

The most optimal explosive combination at this time is the RDX Teslium combo. A really good chemist like myself can produce a small number of devices like this by the 30-35 minute mark using chemfactory mechanics, although mass production is a bit harder and takes longer to pull off. I suspect TaTP has slightly smaller production time requirements. I never wasted time on it myself. 

To my knowledge, I'm the only player who even messes much with the high yield explosive recipes in this game. I'm a little befuddled as to why coders spend their time debuffing any of this when it rarely appears.

**Comment by EdgeLordExe** [comment]
> Experienced chem/explosive manufacturer powergamer here. Perhaps some insight about the current explosive production mechanics in practice would be appreciated.
> 
> For a botany - chemistry partnership to produce a single good nitro bomb within the time of a typical round, a number of requirements need to be satisfied.
> -The botanist and chemist must be blood brothers, or must both have sufficient imagination to understand the multitude of problems that high yield ordnance solves for non antags.
> -The duo must be good at their jobs.
> -The duo must be uninterrupted in their work.
> -Cargo must be competent enough to print small quantities of simple items at their autolathe for those who ask.
> 
> You're wasting your time bothering with nitro grenades. These conditions almost never happen. At the moment they are code bloat.
> 
> The most optimal explosive combination at this time is the RDX Teslium combo. A really good chemist like myself can produce a small number of devices like this at the 30-35 minute mark using chemfactory mechanics, although mass production is a bit harder and takes longer to pull off. I suspect TaTP has slightly smaller production time requirements. I never wasted time on it myself.
> 
> To my knowledge, I'm the only player who even messes much with the high yield explosive recipes in this game. I'm a little befuddled as to why coders spend their time debuffing any of this when it rarely appears.

im just making high-yield explosives harder to make, and at the same time nerfing the easy ones

File: code/modules/reagents/chemistry/recipes.dm [hunk 2]
```diff
@@ -1,23 +1,74 @@
...
+	///Required chemicals that must be present in the container but are not USED.
 	var/list/required_catalysts = new/list()
 
 	// Both of these variables are mostly going to be used with slime cores - but if you want to, you can use them for other things
-	var/required_container = null // the exact container path required for the reaction to happen
-	var/required_other = 0 // an integer required for the reaction to happen
-
-	var/mob_react = TRUE //Determines if a chemical reaction can occur inside a mob
+	/// the exact container path required for the reaction to happen
+	var/required_container = null
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes.dm:18]:
```suggestion
	var/required_container
```

File: code/modules/reagents/chemistry/recipes.dm [hunk 3]
```diff
@@ -1,23 +1,74 @@
...
 
+	///Determines if a chemical reaction can occur inside a mob
+	var/mob_react = TRUE
+	///Required temperature for the reaction to begin
 	var/required_temp = 0
-	var/is_cold_recipe = 0 // Set to 1 if you want the recipe to only react when it's BELOW the required temp.
-	var/mix_message = "The solution begins to bubble." //The message shown to nearby people upon mixing, if applicable
-	var/mix_sound = 'sound/effects/bubbles.ogg' //The sound played upon mixing, if applicable
+	/// Set to TRUE if you want the recipe to only react when it's BELOW the required temp.
+	var/is_cold_recipe = 0
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes.dm:27]:
```suggestion
	var/is_cold_recipe = FALSE
```

**Comment by MarioWizard119** [comment]
Here’s the thing about lavaland geysers

Chemists don’t have a 40 damage infinite ammo gun, and none of their chems work on simplemobs.
Miners are not stupid, and the ones that are are dead within 10 minutes.
What benefit would they have if they cooperated? The miner will get blown sky-high, along with the rest of the station.

The only purpose of plasma oxide is for the exotic stabilizer, which unless the miner and chemist were blood bros or cooperating traitors, would be suspicious as hell. Same scenario if you’re a botanist and you run into a chemist asking for absurd amounts of corn. This adds no depth nor skill requirements for chemists and relies solely on the cooperation, no, stupidity of other departments, unless you count kiting lavaland mobs with a worse disadvantage than ash lizards as skill.

You didn’t just make them hard to make, you made them damn near impossible.

**Comment by ExcessiveUseOfCobblestone** [comment]
Miners cannot look at the contents of a geyser without sci-goggles.

**Comment by EdgeLordExe** [comment]
> Here’s the thing about lavaland geysers
> 
> Chemists don’t have a 40 damage infinite ammo gun, and none of their chems work on simplemobs.
> Miners are not stupid, and the ones that are are dead within 10 minutes.
> What benefit would they have if they cooperated? The miner will get blown sky-high, along with the rest of the station.
> 
> The only purpose of plasma oxide is for the exotic stabilizer, which unless the miner and chemist were blood bros or cooperating traitors, would be suspicious as hell. Same scenario if you’re a botanist and you run into a chemist asking for absurd amounts of corn. This adds no depth nor skill requirements for chemists and relies solely on the cooperation, no, stupidity of other departments, unless you count kiting lavaland mobs with a worse disadvantage than ash lizards as skill.
> 
> You didn’t just make them hard to make, you made them damn near impossible.

just go to lavaland with a spear lmao. If you are that unrobust to not be able to handle a few simplemobs with a spear then you shouldnt be making high-yield explosives

**Comment by EdgeLordExe** [comment]
> Miners cannot look at the contents of a geyser without sci-goggles.

ill add the ability of mesons to see contents of geysers in a diffrent PR, It shouldnt be that hard.

**Comment by EdgeLordExe** [comment]
thanksies cobby!

**Comment by wesoda25** [comment]
> > the fact of the matter is that within chemistry you can't try and say that difficulty is an obstacle because when - not if - people overcome it, they'll just abuse it and kill others unfairly
> 
> god chemistry really needs a rework. It is shallow as fuck.

Late response but yeah, I was just thinking about this the otherday. Chemistry is a terrible system but I don't think its too shallow - its too in depth. There are absolutely way too many chemicals and the problem is that people often need a moderate amount of x chemical, and thus using plumbing is pointless. We need chemistry to not require a massive wiki page for all the chemicals, and instead return to the basics.

**Comment by halitosisman** [comment]
> > > the fact of the matter is that within chemistry you can't try and say that difficulty is an obstacle because when - not if - people overcome it, they'll just abuse it and kill others unfairly
> > 
> > 
> > god chemistry really needs a rework. It is shallow as fuck.
> 
> Late response but yeah, I was just thinking about this the otherday. Chemistry is a terrible system but I don't think its too shallow - its too in depth. There are absolutely way too many chemicals and the problem is that people often need a moderate amount of x chemical, and thus using plumbing is pointless. We need chemistry to not require a massive wiki page for all the chemicals, and instead return to the basics.

The other logical solution is to add a reason for using massive quantities of chemicals. (Which is why every factory building game needs a combat system to fuel demand and vice versa.)
